core/controller/sync.ts

```diff
@@ -1,46 +1,31 @@
 import SyncService from 'core/service/sync';
 import { ON } from 'core/util/decorator';
-import { Block } from 'shared/model/block';
 import { Peer } from 'shared/model/peer';
 import { BaseController } from 'core/controller/baseController';
 import PeerService from 'core/service/peer';
 import RoundService from 'core/service/round';
 import { logger } from 'shared/util/logger';
-import BlockController from 'core/controller/block';
 import PeerRepository from 'core/repository/peer';
-import SharedTransactionRepo from 'shared/repository/transaction';
 import { messageON } from 'shared/util/bus';
 import System from 'core/repository/system';
+import BlockRepository from 'core/repository/block';
+import EventQueue from 'core/repository/eventQueue';
+import { REQUEST_TIMEOUT } from 'core/repository/socket';
 
 export class SyncController extends BaseController {
 
-    @ON('EMIT_REQUEST_COMMON_BLOCKS')
-    emitRequestCommonBlocks(blockData: { id: string, height: number }): void {
-        logger.debug(`[Controller][Sync][emitRequestCommonBlocks] id: ${blockData.id} height: ${blockData.height}`);
-        SyncService.requestCommonBlocks(blockData);
-    }
 
     @ON('REQUEST_COMMON_BLOCKS')
-    checkCommonBlocks(action: { data: { block: { id: string, height: number } }, peer: Peer }): void {
-        const { data, peer } = action;
+    checkCommonBlocks(action: {
+        data: { block: { id: string, height: number } },
+        peer: Peer,
+        requestId: string
+    }): void {
```

Comment:
create type for `{
        const { data, peer } = action;	        data: { block: { id: string, height: number } },
        peer: Peer,
        requestId: string
    }`